File: app/src/main/java/com/shen/xi/android/tut/util/MySimpleImageLoadingListener.java
package com.shen.xi.android.tut.util;

import android.graphics.Bitmap;
import android.view.View;
import android.widget.ImageView;

import com.nostra13.universalimageloader.core.listener.SimpleImageLoadingListener;


public class MySimpleImageLoadingListener extends SimpleImageLoadingListener {

  private final int mMaxWidth;
  private final int mMaxHeight;

  public MySimpleImageLoadingListener(int maxWidth, int maxHeight) {
    mMaxWidth = maxWidth;
    mMaxHeight = maxHeight;
  }

  @Override
  public void onLoadingComplete(String imageUri, View view,
                                Bitmap loadedImage) {

    int imageWidth = loadedImage.getWidth();
    int imageHeight = loadedImage.getHeight();

    boolean overWidth = imageWidth > mMaxWidth;
    boolean overHeight = imageHeight > mMaxHeight;
    if (overHeight || overWidth) {
      loadedImage = Bitmap.createBitmap(loadedImage, 0, 0,
                                        overWidth ? mMaxWidth : imageWidth,
                                        overHeight ? mMaxHeight : imageHeight);
    }
    ((ImageView) view).setImageBitmap(loadedImage);

  }
}

File: app/src/main/scala/com/shen/xi/android/tut/util/MySimpleImageLoadingListener.scala
package com.shen.xi.android.tut.util

import android.graphics.Bitmap
import android.view.View
import android.widget.ImageView
import com.nostra13.universalimageloader.core.listener.SimpleImageLoadingListener

class MySimpleImageLoadingListener(maxWidth: Int, maxHeight: Int, loadingComplete: (String, View, Bitmap) => Unit = null)
  extends SimpleImageLoadingListener {

  override def onLoadingComplete(imageUri: String, view: View, loadedImage: Bitmap): Unit = {
    val imageWidth = loadedImage.getWidth
    val imageHeight = loadedImage.getHeight

    val overWidth = imageWidth > maxWidth
    val overHeight = imageHeight > maxHeight

    if (overHeight || overWidth) {
      val modifiedImage = Bitmap.createBitmap(loadedImage, 0, 0,
        if (overWidth && maxWidth > 0) maxWidth else imageWidth,
        if (overHeight && maxHeight > 0) maxHeight else imageHeight)
      view.asInstanceOf[ImageView].setImageBitmap(modifiedImage)
    } else {
      view.asInstanceOf[ImageView].setImageBitmap(loadedImage)
    }

    if (loadingComplete != null) loadingComplete(imageUri, view, loadedImage)
  }

}
